### PR TITLE
[AP-902][AP-905] Drop pg replication slot in case of full re-sync of a tap.

### PR DESCRIPTION
**Pull request by Samira-El**
## Problem

**AP-905** 
This is a bug in FastSync that allows running Fastsync with tables that aren't configured in the tap with the help of the arg `--tables`.

**AP-902**
When a full resync of the whole tap is triggered, PipelineWise creates a slot if it doesn’t exist.
while the full resync is taking place, if a slot already exists and occupies a lot of disk space, it will grow in size as it’s not consumed from, this can create disk space problems for small but active dbs. 

## Proposed changes

**AP-905** 
Check the list of tables provided via `tables` arg against the set of selected tables in the properties file of the tap. 
Change is in commit

**AP-902** 
Since Full resync is not making use of the slot anyway, and the latter might grow too much, add ability to drop the slot.
That is done automatically by checking the value of arg `tables`, if none is provided then enable the dropping slot by calling a class method `FastSyncTapPostgres.drop_slot`.

## Types of changes

What types of changes does your code introduce to PipelineWise?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] Description above provides context of the change
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Unit tests for changes (not needed for documentation changes)
- [ ] CI checks pass with my changes
- [ ] Bumping version in `setup.py` is an individual PR and not mixed with feature or bugfix PRs
- [ ] Commit message/PR title starts with `[AP-NNNN]` (if applicable. AP-NNNN = JIRA ID)
- [x] Branch name starts with `AP-NNN` (if applicable. AP-NNN = JIRA ID)
- [ ] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions

### CODE REVIEW

File: pipelinewise/fastsync/commons/tap_postgres.py [hunk 1]
```diff
@@ -152,16 +241,7 @@ def fetch_current_log_pos(self):
...
-            self.connection_config['host'],
-            self.connection_config['port'],
-            self.connection_config['user'],
-            self.connection_config['password'],
-            self.connection_config['dbname']
-        )
-        self.primary_host_conn = psycopg2.connect(primary_host_conn_string)
-        # Set connection to autocommit
-        self.primary_host_conn.autocommit = True
+        self.primary_host_conn = self.get_connection(self.connection_config, prioritize_primary=True)
```
Comment (koszti) [inline pipelinewise/fastsync/commons/tap_postgres.py:244]: 🥇 , it connecting to pg was not only duplicated but `'ssl'` option was completely ignored on `primary_host_conn`. Not any more.

File: tests/units/fastsync/assertions.py [hunk 2]
```diff
@@ -153,29 +158,35 @@ def assert_main_impl_should_exit_with_error_on_failure(main_impl: callable,
...
-                        mock_enter.return_value.map.assert_called_once()
+                    with patch(objects_to_mock.full_tap_class_nm) as tap_mock:
+                        tap_mock.return_value.drop_slot.side_effect = None
+
+                        ns = Namespace(**{
+                            'tables': ['table_1', 'table_2', 'table_3', 'table_4'],
+                            'target': 'sf',
+                            'transform': None,
+                            'drop_pg_slot': True,
+                            'tap': {},
```
Comment (koszti) [inline tests/units/fastsync/assertions.py:169]: what is the empty `tap` key is doing here?

Comment (Samira-El) [reply tests/units/fastsync/assertions.py:169]:
```
    if args.drop_pg_slot:
        FastSyncTapPostgres.drop_slot(args.tap)
```
It's to mock `args.tap` that's done when `drop_pg_slot` is `True`

**Review by koszti — APPROVED** [review]

**Comment by koszti** [comment]
super nice

**Pull request merged**